stripe/src/main/java/com/stripe/android/PaymentSession.java

```diff
@@ -0,0 +1,204 @@
+     * otherwise {@code false}
+     */
+    public boolean handlePaymentData(int requestCode, int resultCode, @NonNull Intent data) {
+        if (resultCode == Activity.RESULT_CANCELED) {
+            return false;
+        } else if (resultCode == Activity.RESULT_OK) {
+            switch (requestCode) {
+                case PAYMENT_METHOD_REQUEST:
+                    fetchCustomer();
+                    return true;
+                default:
+                    break;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Initialize the PaymentSession with a {@link PaymentSessionListener} to be notified of
+     * data changes.
+     *
+     * @param listener a {@link PaymentSessionListener} that will receive notifications of changes
+     *                 in payment session status, including networking status
+     * @return {@code true} if the PaymentSession is initialized, {@code false} if a state error
+     * occurs. Failure can only occur if there is no initialized {@link CustomerSession}.
+     */
+    public boolean init(@NonNull PaymentSessionListener listener) {
+        return init(listener, null);
+    }
+
+    /**
+     * Initialize the PaymentSession with a {@link PaymentSessionListener} to be notified of
+     * data changes.
+     *
+     * @param listener a {@link PaymentSessionListener} that will receive notifications of changes
+     *                 in payment session status, including networking status
+     * @param savedInstanceState a {@link Bundle} containing the saved state of a PaymentSession
+     *                           that was stored in {@link #savePaymentSessionInstanceState(Bundle)}
+     * @return {@code true} if the PaymentSession is initialized, {@code false} if a state error
+     * occurs. Failure can only occur if there is no initialized {@link CustomerSession}.
+     */
+    public boolean init(
+            @NonNull PaymentSessionListener listener,
+            @Nullable Bundle savedInstanceState) {
+
+        // Checking to make sure that there is a valid CustomerSession -- the getInstance() call
+        // will throw a runtime exception if none is ready.
+        try {
+            CustomerSession.getInstance().addProductUsageTokenIfValid(TOKEN_PAYMENT_SESSION);
+        } catch (IllegalStateException illegalState) {
+            mPaymentSessionListener = null;
+            return false;
+        }
+
+        mPaymentSessionListener = listener;
+        if (savedInstanceState != null) {
+            PaymentSessionData data =
+                    savedInstanceState.getParcelable(PAYMENT_SESSION_DATA_KEY);
+            if (data != null) {
+                mPaymentSessionData = data;
+            }
+        }
+        fetchCustomer();
+        return true;
+    }
+
+    /**
+     * Launch the {@link PaymentMethodsActivity} to allow the user to select a payment method,
+     * or to add a new one.
+     */
+    public void selectPaymentMethod() {
```

Comment:
maybe we should name this `start/presentPaymentMethodsActivity`? `selectPaymentMethod` sounds like you're selecting a particular payment method.

(in the iOS SDK this is called `presentPaymentMethodsViewController`)

Reply:
Sure - we can make this update